### PR TITLE
Add Lock to multiprocessing

### PR DESCRIPTION
**Pull request by clavedeluna**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes https://github.com/PyCQA/pylint/issues/3313

### CODE REVIEW

**Comment by clavedeluna** [comment]
Hi maintainers, I'm attempting to fix https://github.com/PyCQA/pylint/issues/3313 . While this change no longer emits `testing.py:4: [E1101(no-member), ] Instance of 'SyncManager' has no 'Lock' member; maybe 'RLock'?` I'm now getting a ` Too many positional arguments for function call (too-many-function-args)` on that line. Trying to investigate and think it's likely due to [this line](https://github.com/python/cpython/blob/53d9cd95cd91f1a291a3923acb95e0e86942291a/Lib/threading.py#L108) . What else can I do here?

**Review by DanielNoord — APPROVED** [review]

**Review by DanielNoord — DISMISSED** [review]
Can't you reference the private import instead?

File: astroid/brain/brain_multiprocessing.py
Comment (Pierre-Sassoulas) [inline astroid/brain/brain_multiprocessing.py:81]:
```suggestion
        Lock = multiprocessing.synchronize.Lock()
```
Would this work ? I'm basing this suggestion on https://docs.python.org/3/library/multiprocessing.html#multiprocessing.Lock

Comment (clavedeluna) [reply astroid/brain/brain_multiprocessing.py:81]: Yay this indeed works perfectly! Thank you!

**Comment by Pierre-Sassoulas** [comment]
The ``tests/test_import_graph.py::test_checker_dep_graphs`` test break in pylint with this version of astroid but it seems to also break with the current ``main`` branch of astroid, so...

File: tests/unittest_brain.py
Comment (cdce8p) [inline tests/unittest_brain.py:714]:
```suggestion
        lock = manager.Lock()
```

File: tests/unittest_brain.py
Comment (cdce8p) [inline tests/unittest_brain.py:741]:
```suggestion
        lock_name = "threading.lock"
```

**Review by cdce8p — CHANGES_REQUESTED** [review]

**Comment by clavedeluna** [comment]
Hi @cdce8p your review (one part outdated) is a bit confusing to me, I can't figure out if your suggestion of using `threading` is what we want or not, given we're focusing on `multiprocessing` module. Maybe if you add a commit with what you're suggesting I can test it and verify the original test case does work.

**Comment by cdce8p** [comment]
> Maybe if you add a commit with what you're suggesting I can test it and verify the original test case does work.

Sorry for the confusion, that wasn't may intention. Pushed the commit. It's basically your original proposal but was a small change in `brain_threading.py` to make it work.

I think for the `SyncManager` it does make sense to use `Lock = threading.Lock` as this is closest to the ones already there.
https://github.com/python/cpython/blob/v3.11.1/Lib/multiprocessing/managers.py#L1214-L1215

**Comment by clavedeluna** [comment]
@cdce8p ahh perfect! I got it now and it does indeed work 👏

**Comment by cdce8p** [comment]
> @cdce8p ahh perfect! I got it now and it does indeed work 👏

Great! Could you add a short changelog entry for `2.13.4`. After that we can merge it.

**Review by cdce8p — APPROVED** [review]
Looks good! Thanks @clavedeluna 👍🏻

**Pull request merged**
